webxr/headless/mod.rs

```diff
@@ -70,59 +75,68 @@ impl MockDiscovery for HeadlessMockDiscovery {
         init: MockDeviceInit,
         receiver: Receiver<MockDeviceMsg>,
     ) -> Result<Box<dyn Discovery>, Error> {
+        let viewer_origin = init.viewer_origin.clone();
+        let floor_transform = init.floor_origin.inverse();
+        let views = init.views.clone();
+        let data = HeadlessDeviceData {
+            floor_transform,
+            viewer_origin,
+            views,
+            inputs: vec![],
+            events: Default::default(),
+            quitter: None,
+            disconnected: false,
+        };
+        let data = Arc::new(Mutex::new(data));
+        let data_ = data.clone();
+
+        thread::spawn(move || {
+            run_loop(receiver, data_);
```

Comment:
OK I am confused by this, why is `HeadlessMockDiscovery` spinning up a thread?

Reply:
We need to be able to request sessions to the same device multiple times. The device is longer lived -- if a _session_ ends, the device sticks around, but if the _device_ disconnects you have to rerun simulatedeviceconnection.

Given that simulateDeviceConnection spawns a Discovery, the model I'm following is that each Discovery is a single device (which is forever lost once disconnected, though it stays in the array as an inert Discovery object)

Reply:
Hmm, that's a different model than I was thinking of, which is that each `Device` object is only as long-lived as its session, so it made sense to have it owned by the `SessionThread`. If nothing else, that means the name isn't ideal, as `Device` should be `ActiveDevice`?

Reply:
This won't work, because you need to be able to persist mock device state between sessions.

Reply:
You're right, the `Device` trait is the session, and we could rename it. But we still need a separate event loop here.